### PR TITLE
Fix `data_parallelism` related code in v2

### PR DESCRIPTION
**Pull request by buptchan**
Fix `data_parallelism` related code in several policies.

## Type of Change

- [x] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [x] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [x] Linux
- Python version:
  - [ ] 3.6
  - [x] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [x] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: maro/rl/policy_v2/policy_base.py [hunk 1]
```diff
@@ -212,22 +213,23 @@ def get_batch_loss(self, batch: dict, explicit_grad: bool = False) -> object:  #
         """
         pass
 
-    @abstractmethod
     def data_parallel(self, *args, **kwargs) -> None:
         """"Initialize a proxy in the policy, for data-parallel training.
         Using the same arguments as `Proxy`."""
-        pass
+        self.task_queue_client = TaskQueueClient()
```
Comment (lihuoran) [inline maro/rl/policy_v2/policy_base.py:219]: 1. Instance attribute `task_queue_client` defined outside `__init__ `
2. Should we use `_task_queue_client` if it is an internal variable?

File: maro/rl/policy_v2/policy_base.py [hunk 2]
```diff
@@ -212,22 +213,23 @@ def get_batch_loss(self, batch: dict, explicit_grad: bool = False) -> object:  #
...
     def data_parallel_with_existing_proxy(self, proxy: Proxy) -> None:
         """"Initialize a proxy in the policy with an existing one, for data-parallel training."""
-        self._proxy = proxy
+        self.task_queue_client = TaskQueueClient()
+        self.task_queue_client.set_proxy(proxy)
 
     def exit_data_parallel(self) -> None:
-        if self._proxy is not None:
-            self._proxy.close()
+        if hasattr(self, "task_queue_client"):
```
Comment (lihuoran) [inline maro/rl/policy_v2/policy_base.py:228]: Could we avoid using `hasattr`? In my personal opinion, the better way is create a private attribute `_task_queue_client` with type `Optional[TaskQueueClient]`, and call it if it is not `None`.

File: maro/rl/policy_v2/maac.py [hunk 3]
```diff
@@ -239,7 +239,7 @@ def get_batch_loss(self, batch: dict, explicit_grad: bool = False) -> dict:
     def data_parallel(self, *args, **kwargs) -> None:
```
Comment (lihuoran) [inline maro/rl/policy_v2/maac.py:239]: Maybe we should remove `data_parallel()` here?

File: maro/rl/policy_v2/policy_base.py (hunk 1 above)
Comment (buptchan) [reply maro/rl/policy_v2/policy_base.py:219]: The `TaskQueueClient` will be created only in the `data_parallel` mode, and another method `data_parallel_with_existing_proxy` met the same problem.

File: maro/rl/policy_v2/maac.py (hunk 3 above)
Comment (buptchan) [reply maro/rl/policy_v2/maac.py:239]: I leave all the changes of `maac.py` in another pull request since this one is temporary.

**Review by ysqyang — APPROVED** [review]

**Pull request merged**